synapse/app/homeserver.py

```diff
@@ -197,25 +205,31 @@ def start_listening(self):
                 bind_addresses = listener["bind_addresses"]
 
                 for address in bind_addresses:
-                    reactor.listenTCP(
-                        listener["port"],
-                        manhole(
-                            username="matrix",
-                            password="rabbithole",
-                            globals={"hs": self},
-                        ),
-                        interface=address
-                    )
+                    try:
+                        reactor.listenTCP(
+                            listener["port"],
+                            manhole(
+                                username="matrix",
+                                password="rabbithole",
+                                globals={"hs": self},
+                            ),
+                            interface=address
+                        )
+                    except error.CannotListenError as e:
+                        check_bind_error(e, address, bind_addresses)
```

Comment:
I'm not sure it makes sense to apply the same logic to the rabbithole or replication listeners, not least because if you're binding them to the wildcard addresses, you're probably doing it wrong.